MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/XmpParser.java

```diff
@@ -63,19 +64,21 @@ private static int indexOf(byte[] arr, byte[] seq, int start) {
     /**
      * Returns the metadata of the Motion Photo file.
      */
+    @Nullable
     public static XMPMeta getXmpMetadata(String filename) throws IOException, XMPException {
-        FileInputStream in = new FileInputStream(filename);
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try (FileInputStream in = new FileInputStream(filename)) {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            ByteStreams.copy(in, out);
 
-        copy(filename, in, out);
-        byte[] fileData = out.toByteArray();
+            byte[] fileData = out.toByteArray();
 
-        int openIdx = indexOf(fileData, OPEN_ARR, 0);
-        if (openIdx >= 0) {
-            int closeIdx = indexOf(fileData, CLOSE_ARR, openIdx + 1) + CLOSE_ARR.length;
+            int openIdx = indexOf(fileData, OPEN_ARR, 0);
```

Comment:
I looked this up, but turns out you can also use Bytes.indexOf(fileData, OPEN_ARR) from Guava :D

https://guava.dev/releases/19.0/api/docs/com/google/common/primitives/Bytes.html#indexOf(byte[],%20byte[])